apps/omg_eth/lib/omg_eth/client.ex

```diff
@@ -29,13 +29,21 @@ defmodule OMG.Eth.Client do
     end
   end
 
-  @spec node_ready() :: :ok | {:error, :geth_still_syncing | :geth_not_listening}
-  @spec node_ready(module()) :: :ok | {:error, :geth_still_syncing | :geth_not_listening}
+  @spec node_ready() :: :ok | {:error, :geth_still_syncing | :geth_not_listening | :closed}
+  @spec node_ready(module()) :: :ok | {:error, :geth_still_syncing | :geth_not_listening | :closed}
   def node_ready(client \\ Ethereumex.HttpClient) do
     case client.eth_syncing() do
```

Comment:
I think hackney can return  additional types of errors
https://github.com/benoitc/hackney/blob/b4a48ed5977c337ce0f761ad59e32ec634ebcbc8/src/hackney_stream.erl#L34
what do you think about logging unknown errors?

```elixir
{:error, _} = unknow_error -> Logger.error(...
```

Reply:
why would we log them manually? a crash log produces an error log...

Reply:
this function is expected to crash?

the additional clause will catch an unmatched error

Reply:
well the question is - what can you do with a response you're not able to handle on the caller? the OTP way is to let it explode and let the process (the caller) restart and try again.

Reply:
Some errors are handled, some errors are ok to crash

I think unknown errors can also be handled

Reply:
so we going with the some error to crash way then?